pipeline-scripts/commonlib.groovy

```diff
@@ -593,31 +593,36 @@ def invalidateAwsCache(s3_path) {
-                    // 2. On the second pass, include only the repomd.xml.
-                    shell(script: "aws s3 sync ${opts} --exclude '*' --include '*/repomd.xml' ${local_dir} s3://art-srv-enterprise${s3_path}")
-                    if (remove_old) {
-                        // For most repos, clean up the old rpms so they don't grow unbounded. Specify remove_old=false
-                        // to prevent this step.
-                        // Otherwise:
-                        // 3. Everything should be sync'd in a consistent way -- delete anything old with --delete.
-                        shell(script: "aws s3 sync ${opts} --delete ${local_dir} s3://art-srv-enterprise${s3_path}")
+        for secretKeyId in ('s3-art-srv-enterprise', 's3-art-srv-enterprise-cloudflare') {
+            withCredentials([
+            aws(credentialsId: secretKeyId, accessKeyVariable: 'AWS_ACCESS_KEY_ID', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'),
+            string(credentialsId: "${secretKeyId}-endpoint", variable: 'AWS_ENDPOINT_URL')]) {
+                retry(3) {
+                    timeout(time: timeout_minutes, unit: 'MINUTES') { // aws s3 sync has been observed to hang before
+                        // Sync is not transactional. If we update repomd.xml before files it references are populated,
+                        // users of the repo will get a 404. So we run in three passes:
+                        // 1. On the first pass, exclude files like repomd.xml and do not delete any old files. This ensures that we  are only adding
+                        // new rpms, filelist archives, etc.
+                        def opts = '--no-progress --exact-timestamps'
+                        if (dry_run) opts += ' --dryrun'
+                        shell(script: "aws s3 sync ${opts} --exclude '*/repomd.xml' ${local_dir} s3://art-srv-enterprise${s3_path}") // Note that s3_path has / prefix.
+                        // 2. On the second pass, include only the repomd.xml.
+                        shell(script: "aws s3 sync ${opts} --exclude '*' --include '*/repomd.xml' ${local_dir} s3://art-srv-enterprise${s3_path}")
+                        if (remove_old) {
+                            // For most repos, clean up the old rpms so they don't grow unbounded. Specify remove_old=false
+                            // to prevent this step.
+                            // Otherwise:
+                            // 3. Everything should be sync'd in a consistent way -- delete anything old with --delete.
+                            shell(script: "aws s3 sync ${opts} --delete ${local_dir} s3://art-srv-enterprise${s3_path}")
+                        }
                     }
                 }
-            }
-            if (issue_cloudfront_invalidation) {
-                invalidateAwsCache(s3_path)
+                if (issue_cloudfront_invalidation) {
```

Comment:
This can be outside the loop. There is no caching with CloudFront on CloudFlare.

Reply:
If we keep outside, then we would have to use `withCredentials` again right, so added `secretKeyId == 's3-art-srv-enterprise'` to the if